src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -215,29 +167,100 @@ export class MeetingManager implements AudioVideoObserver {
     this.audioVideoObservers = {};
   }
 
-  async join({
-    meetingInfo,
-    attendeeInfo,
-    deviceLabels = DeviceLabels.AudioAndVideo,
-    eventController,
-    meetingManagerConfig,
-  }: MeetingJoinData) {
-    if (meetingManagerConfig) {
-      this.meetingManagerConfig = meetingManagerConfig;
-    }
-    this.configuration = new MeetingSessionConfiguration(
-      meetingInfo,
-      attendeeInfo
+  async join(
+    meetingSessionConfiguration: MeetingSessionConfiguration,
+    options?: MeetingManagerJoinOptions
+  ): Promise<void> {
+    const {
+      deviceLabels,
+      eventController,
+      logger,
+      enableWebAudio,
+      activeSpeakerPolicy,
+    } = this.parseJoinParams(meetingSessionConfiguration, options);
+    this.meetingSessionConfiguration = meetingSessionConfiguration;
+    this.meetingId = this.meetingSessionConfiguration.meetingId;
+    this.logger = logger;
+
+    const deviceController = new DefaultDeviceController(logger, {
+      enableWebAudio: enableWebAudio,
+    });
+    this.meetingSession = new DefaultMeetingSession(
+      meetingSessionConfiguration,
+      logger,
+      deviceController,
+      eventController,
     );
 
-    this.meetingRegion = meetingInfo.MediaRegion;
-    this.meetingId = this.configuration.meetingId;
+    this.audioVideo = this.meetingSession.audioVideo;
 
     if (eventController) {
-      this.eventController = eventController;
+      eventController.addObserver(this.eventDidReceiveRef);
+    } else {
+      this.meetingSession.eventController.addObserver(this.eventDidReceiveRef);
+    }
+
+    this.setupAudioVideoObservers();
+    this.setupDeviceLabelTrigger(deviceLabels);
+    await this.listAndSelectDevices(deviceLabels);
+    this.publishAudioVideo();
+    this.setupActiveSpeakerDetection(activeSpeakerPolicy);
+    this.meetingStatus = MeetingStatus.Loading;
+    this.publishMeetingStatus();
+  }
+
+  private parseJoinParams(
+    meetingSessionConfiguration: MeetingSessionConfiguration,
+    options: MeetingManagerJoinOptions | undefined
+  ) {
+    let deviceLabels: DeviceLabels | DeviceLabelTrigger;
+    let eventController: EventController | undefined;
+    let logLevel: LogLevel;
+    let postLoggerConfig: PostLoggerConfig | undefined;
+    let logger: Logger;
+    let enableWebAudio: boolean;
+    let activeSpeakerPolicy: ActiveSpeakerPolicy;
+
+    // TODO: The following if blocks can be replaced with ?? null-coalescing operator when the SDK supportes es 2020.
+    if (options?.deviceLabels) {
+      deviceLabels = options?.deviceLabels;
+    } else {
+      deviceLabels = DeviceLabels.AudioAndVideo;
+    }
```

Comment:
I think these can be simplified using `||`.
For example:
```js
deviceLabels = options?.deviceLabels || DeviceLabels.AudioAndVideo;
```

`||` is similar as `??`.


>The [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null)ish coalescing operator (??) is a logical operator that returns its right-hand side operand when its left-hand side operand is null or [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined), and otherwise returns its left-hand side operand.
>
>This can be seen as a special case of the logical OR (||) operator, which returns the right-hand side operand if the left operand is any [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value, not only null or undefined.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator